### PR TITLE
ci: use github token permissions instead of personal access tokens

### PR DESCRIPTION
**Pull request by nunofgs**
### Description

Replaces PATs currently being used for various CI actions with `GITHUB_TOKEN` and respective [permissions](https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/).

### Note to reviewers
<details>
  <summary>The `deploy-cocoapods` and `deploy-git-tag` workflows were merged into a single workflow.</summary>
  <br/>
  <ul>
   <li>This was required because events generated by the use of `GITHUB_TOKEN` <a href="https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow">will not trigger</a> other workflows.</li>
  <li>
  This will be the new flow:
  Screenshot 2023-02-03 at 10 44 17
  </li>
</details>

<details>
  <summary>The `action-semantic-release-justwork` step was removed.</summary>
  <br/>
  <ul>
   <li>This is no longer necessary since we're not updating long-running tags anymore. (such as `v1`, `v2`, etc.)</li>
  </ul>
</details>

<details>
  <summary>Pull requests are no longer created as `develop` branch is being phased out.</summary>
  <br/>
  <ul>
   <li>Previously this was only done if the automatic merge failed.</li>
  </ul>
</details>

Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-ios/blob/develop/docs/dev-notes/GIT-WORKFLOW.md). 
- [x] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request. 
- [x] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1). 
- [x] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After pull request is approved, and you determine it's ready **add the label "Ready to merge"** to the pull request. A bot will *squash and merge* the pull request for you after the label is added.

### CODE REVIEW

File: .github/workflows/promote.yml [hunk 1]
```diff
@@ -12,10 +15,7 @@ jobs:
         uses: levibostian/action-promote-semantic-release@v1
         with:
           sequence: "develop,alpha,beta,main"
-          githubToken: ${{ secrets.WRITE_ACCESS_BOT_TOKEN }}
-          # Find private email address that github uses for CI account at: https://github.com/settings/emails
-          gitName: "Ami CI"
-          gitEmail: "21171259+ami-ci@users.noreply.github.com"
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
```
Comment (levibostian) [inline .github/workflows/promote.yml:18]: In our call together, we talked about how you were experiencing issues with the CI server not being able to make git commits. By removing `gitName` and `gitEmail`, doesn't that mean that we have this problem again?

**Review by levibostian — APPROVED** [review]

**Comment by levibostian** [comment]
The 2 failing checks [are a known issue](https://github.com/customerio/issues/issues/9279) and should not block merging this PR.

**Comment by levibostian** [comment]
The setting you mentioned in the description is enabled for this repo.

**Comment by levibostian** [comment]
Do not merge until [decision is made](https://github.com/customerio/issues/issues/9306). If we modify our deployment strategy or not. 

If we do, then this PR and other PRs in the other mobile repos can be simplified to not need actions to support keeping branches in sync.

File: .github/workflows/pr-helper.yml [hunk 2]
```diff
@@ -4,6 +4,9 @@ on:
   pull_request:
     types: [opened, reopened, edited, synchronize, labeled]
 
+permissions:
+  pull-requests: write # Write access needed to create a comment.
```
[the remark this answers is not in the document]
Comment (nunofgs) [reply .github/workflows/pr-helper.yml]: Unfortunately doesn't work (tried it before 😢). Here's a test run I just triggered: https://github.com/customerio/customerio-ios/actions/runs/4102666719/jobs/7075906418

The action creates comments [here](https://github.com/levibostian/action-semantic-pr/blob/d1d7ed7b1a4a77cf9b85d47b154a02939fd6f2e1/app/index.ts#L52) and [here](https://github.com/levibostian/action-semantic-pr/blob/d1d7ed7b1a4a77cf9b85d47b154a02939fd6f2e1/app/index.ts#L99) and pull-requests `write` is unfortunately the only permission that can allow that.

(I would much rather prefer a `comments: write`... hopefully Github adds it)

File: .github/workflows/pr-review.yml [hunk 3]
```diff
@@ -2,6 +2,11 @@ name: PR review
 
 on: [pull_request]
 
+permissions:
+  contents: read # Access needed to checkout the repo.
+  issues: read # Needed to retrieve commits on a pull request.
+  pull-requests: write # Write access needed to create a comment.
```
[the remark this answers is not in the document]
Comment (nunofgs) [reply .github/workflows/pr-review.yml:8]: Same deal as [before](https://github.com/customerio/customerio-ios/pull/261#discussion_r1097217294) 😢

File: .github/workflows/pr-helper.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (levibostian) [reply .github/workflows/pr-helper.yml:8]: Thanks for adding a comment as to why that permission is needed. Thanks for trying it for us, too!

**Comment by Shahroz16** [comment]
Thanks, Levi for taking lead and removing branches, it would be super helpful if you can also mention these on slack (team-mobile?) so everyone making a PR knows which branch to target. This comment can get lost in tickets.

File: .github/workflows/deploy-sdk.yml [hunk 4]
```diff
@@ -0,0 +1,215 @@
+name: Deploy SDK
+
+on:
+  push:
+    branches: [main, v1] # all branches where deployments currently occur. Make sure this list matches list of branches in  `.releaserc` file.
```
Comment (nunofgs) [inline .github/workflows/deploy-sdk.yml:5]: @levibostian We still have `alpha` and `beta` in the .releaserc file. Should those be changed in this PR?

**Pull request merged**